src/devices/ServoMotor/ServoMotor.cs

```diff
@@ -0,0 +1,133 @@
+        public ServoMotor(
+            int chip,
+            int channel,
+            int maximumAngle = 180,
+            int minimumPulseWidthMicroseconds = 1_000,
+            int maximumPulseWidthMicroseconds = 2_000)
+            : this(maximumAngle, minimumPulseWidthMicroseconds, maximumPulseWidthMicroseconds)
+        {
+            _pwmChannel = PwmChannel.Create(chip, channel, 50);  // Default 50Hz
+        }
+
+        private static double CalculateServoAngleSlope(int maximumAngle, int minimumPulseWidthMicroseconds, int maximumPulseWidthMicroseconds)
+        {
+            if (maximumAngle < 0 || maximumAngle > 360)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maximumAngle), maximumAngle, "Value must be between 0 and 360.");
+            }
+
+            if (minimumPulseWidthMicroseconds < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(minimumPulseWidthMicroseconds), minimumPulseWidthMicroseconds, "Value must not be negative.");
+            }
+
+            if (maximumPulseWidthMicroseconds < minimumPulseWidthMicroseconds)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maximumPulseWidthMicroseconds), maximumPulseWidthMicroseconds, $"Value must be greater than or equal to {minimumPulseWidthMicroseconds}.");
+            }
+
+            // slope = (y2 - y1) / (x2 - x1).  x1 is 0 in this case, so it is not subtracted.
+            double servoAngleSlope = (double)(maximumPulseWidthMicroseconds - minimumPulseWidthMicroseconds) / maximumAngle;
+            return servoAngleSlope;
+        }
+
+        /// <summary>
+        /// Starts the servo motor.
+        /// </summary>
+        public void Start() => _pwmChannel.Start();
+
+        /// <summary>
+        /// Stops the servo motor.
+        /// </summary>
+        public void Stop() => _pwmChannel.Stop();
+
+        /// <summary>
+        /// Writes an angle to the servo motor.
+        /// </summary>
+        /// <param name="angle">The angle to write to the servo motor.</param>
+        public void WriteAngle(int angle)
+        {
+            if (angle < 0 || angle > _maximumAngle)
+            {
+                throw new ArgumentOutOfRangeException(nameof(angle), angle, $"Value must be between 0 and {_maximumAngle}.");
+            }
+
+            WritePulseWidth((int)(_servoAngleSlope * angle + _minimumPulseWidthMicroseconds));
+        }
+
+        /// <summary>
+        /// Writes a pulse width to the servo motor.
+        /// </summary>
+        /// <param name="microseconds">The pulse width, in microseconds, to write to the servo motor.</param>
+        public void WritePulseWidth(int microseconds)
+        {
+            double dutyCyclePercentage = (double)microseconds / 1_000_000 * _pwmChannel.Frequency; // Convert to seconds 1st.
```

Comment:
would it make sense to keep the time in units so that conversion wouldn't be necessary?

Reply:
As far as how the code works, you pass in a value on how long you want the pulse width to be during the cycle period.  This usually ranges from 0.001s/1ms/1000us to 0.002s/2ms/2000us.

This is sort of a bigger discussion to keep consistent (at least with some of the PWM stuff) as units of time will vary how datasheets reference based on the hardware, lower-level APIs, etc.  For example, Linux uses Nanoseconds, servo motor datasheets reference pulse width ranges in microseconds, servo tutorials explain with milliseconds.  Blame, why can't the world be in sync?!?! 😄 

I thought about using the pulse width parameter in seconds as a `double`, but might confuse some users.  Use 900 microseconds as an example.

```csharp
// In microseconds as int.
public void WritePulseWidth(900) { ... }

// In seconds as double.
public void WritePulseWidth(0.000900) { ... }
```

Using seconds and decimal places might cause a user to miss the correct period location compared to using the value provided by datasheet in microseconds.

Long story short, I can change to use seconds as a double and remove conversion if you think that is a better approach.   Still gives same result.

```console
50Hz; 0.020s or 20000us
Setting pulse width of 0.0015s or 1500us

double dutyCyclePercentage = (double)microseconds / 1_000_000 * _pwmChannel.Frequency;

Seconds
0.075 = 0.0015 * 50

Microseconds
0.075 = (double)1500 / 1_000_000 * 50
```

Reply:
if you call this relativeAngle which is value from 0 to 1 and allow to modify MinPulseWidth and MaxPulseWidth and AngleRange (and possibly MinAngle) then I feel like those helpers won't be longer necessary and everything stays clear on how it works. Make sure to match names with whatever manufacturers use